src/backend/optimizer/plan/setrefs.c

```diff
@@ -419,6 +419,13 @@ set_plan_references(PlannerInfo *root, Plan *plan)
 	{
 		PlanRowMark *rc = lfirst_node(PlanRowMark, lc);
 
+		/*
+		 * If the query has several ModifyTable nodes, which are reference 
+		 * the same RTE, we don't need to adjust the rowmark more than once.
+		 * For this case we include only distinct rowmarks in the finalrowmarks,
+		 * referencing to repeated more than once. 
```

Comment:
Let's avoid writing about ModifyTable node case because the code below is executed for all possible cases